src/core/net/dhcp6_client.cpp

```diff
@@ -124,20 +122,20 @@ void Dhcp6Client::UpdateAddresses(void)
         doesAgentExist = true;
         found          = false;
 
-        for (size_t i = 0; i < OT_ARRAY_LENGTH(mIdentityAssociations); i++)
+        for (IdentityAssociation &eachIa : mIdentityAssociations)
```

Comment:
```suggestion
        for (IdentityAssociation &idAssociation : mIdentityAssociations)
```

Reply:
`ia` is already used so I used `eachIa` here.